src/queryBuilder.ts

```diff
@@ -32,6 +32,7 @@ export type GenerateQueryOptions<T extends object> = {
 	orderBy?: OrderBy<T> | OrderBy<T>[];
 	data?: Partial<T>;
 	upsertOnlyUpdateData?: Partial<T>;
+	uniqueColumn?: string;
```

Comment:
Have you run this code against D1 to confirm it works?

Reply:
I think the type can be a little more strict but otherwise I have tested this on a D1 preview and it works fine.